### ISSUE TITLE
32-bit build warnings [salt() returned misaligned pointer]

### ISSUE THREAD

**Issue by claudioandre-br** [openwall/john#3677]
```
Testing: AndroidBackup [PBKDF2-SHA1 128/128 NEON 4x AES]... (4xOMP) Warning: salt() returned misaligned pointer
PASS
Testing: andOTP [SHA256 32/32]... (4xOMP) Warning: salt() returned misaligned pointer
PASS
Testing: bitshares, BitShares Wallet [SHA-512 64/32]... (4xOMP) Warning: salt() returned misaligned pointer
PASS
Testing: radius, RADIUS authentication [MD5 32/32]... (4xOMP) Warning: salt() returned misaligned pointer
PASS
```

A fix can be seen in f7132b346c43443b5ac63372bd57b647d3332d3f

**Comment by solardiz** [comment]
I think f7132b346c43443b5ac63372bd57b647d3332d3f is more of a workaround than a fix. Ideally, we'd define the types such that the compiler provides proper alignment without us having to ask for it. Perhaps this would involve use of `union` types. We'd also avoid strict aliasing rules violations then, which I guess we have now.

**Comment by magnumripper** [comment]
These were fixed in #3934
